exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SessionOptionManager.java

```diff
@@ -46,6 +47,33 @@ public SessionOptionManager(final OptionManager systemOptions, final UserSession
   }
 
   @Override
+  public void deleteOption(final String name, final OptionValue.OptionType type) {
+    if (type == OptionValue.OptionType.SESSION) {
+      try { // ensure option exists
+        SystemOptionManager.getValidator(name);
+      } catch (final IllegalArgumentException e) {
+        throw UserException.validationError()
+          .message(e.getMessage())
+          .build(logger);
+      }
+      options.remove(name);
+      shortLivedOptions.remove(name);
+    } else {
+      fallback.deleteOption(name, type);
```

Comment:
so if I change an option both at the _session_ and _system_ levels, then I _reset_ the same option at the _system_ level it will leave it changed at the _session_ level. Is this expected ?